### ISSUE TITLE
Split up Admin page into groups

### ISSUE THREAD

**Issue by blizzz** [nextcloud/server#712]
We want to have the huge admin page split up in groups. 

Goals:
* avoid to have a single huge, long page
* group different settings reasonable (e.g. everything related to sharing & federation)
* make sure specific settings have full pages (like Log, LDAP etc)
* avoid very small setting pages by another group "Additional settings"

Long term considerations:
* integrate personal settings

Solution:
* anything that has an admin settings page will register a Controller. A group¹ parameter is required.
* legacy support: group apps that do not register a controller into "Additional Settings"
* downside: we need to adjust as many apps as possible and provide developer documentation quickly

¹also requires a way to register groups. Probably another controller to register new groups and have "core" groups pre-defined (sharing, logging, background jobs, …)

@jancborchardt @nickvergessen

**Comment by jancborchardt** [comment]
Good summary. I only adjusted details (added federation, Logs and LDAP as specific examples).

**Comment by mar1u5** [comment]
Maybe we could also integrate @icewind1991's logreader (https://github.com/icewind1991/logreader) as the default one into its own page. Now we would have space for it 😉 What do you think?

**Comment by blizzz** [comment]
@mar1u5 this crossed my mind when I looked into the Logging section today. I would find it useful, however this would be out of scope of this issue (and the PR #796 is already huge).

**Comment by icewind1991** [comment]
Once #796 is done I'll adjust the logreader to register itself in the admin page, we can then look into shipping it on default

**Comment by MorrisJobke** [comment]
This is merged and backported to stable10
